### PR TITLE
LICENSE changes

### PR DESCRIPTION
**Pull request by Aquafina-water-bottle**
# Changes
- Updated LICENSE to full license
- Added Yomitan 2023 to all license headers, and under ext/legal.html
- Updated `.eslintrc.json` with the new header check logic

# ESLint config change explanation
The eslint file requires us to change the once beautiful array of lines into a gigantic regex string, because the extension does not allow us to specify newlines within lines in an array[^1]. For example, the following will always error:

[^1]:
    「In the 2 argument form the first must be either `"block"` or `"line"` to indicate what style of comment should be used. The second is either a string (including newlines) of the comment, or an array of each line of the comment.」 Taken from the [official documentation](https://github.com/Stuk/eslint-plugin-header#2-arguments).

```
{"pattern": "Line1\nLine2"},
"Line3"
```

This detail is important, because we want to test for an optional "Yomichan Authors" secondary line. For example, both examples below should not error:

> Example 1:
> ```
> Line1
> Line2
> Line3
> ```
>
> Example 2:
> ```
> Line1
> Line3
> ```

However, this transforms our required regex check into:

```
{"pattern": "Line1(\nLine2)?"},
"Line3"
```

and since a newline must be present in the regex, this will also error no matter the input.

The only way I found to deal with this is by changing the array to one large regex string.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -169,24 +169,13 @@
...
-                    " * along with this program.  If not, see <https://www.gnu.org/licenses/>.",
-                    " "
-                ]]
+                "header/header": [
+                    "error",
+                    "block",
+                    {
+                        "pattern": " \\* Copyright \\(C\\) (\\d+-)?2023  Yomitan Authors(\n \\* Copyright \\(C\\) (\\d+-)?2022  Yomichan Authors)?\n \\*\n \\* This program is free software: you can redistribute it and/or modify\n \\* it under the terms of the GNU General Public License as published by\n \\* the Free Software Foundation, either version 3 of the License, or\n \\* \\(at your option\\) any later version\\.\n \\*\n \\* This program is distributed in the hope that it will be useful,\n \\* but WITHOUT ANY WARRANTY; without even the implied warranty of\n \\* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE\\.  See the\n \\* GNU General Public License for more details\\.\n \\*\n \\* You should have received a copy of the GNU General Public License\n \\* along with this program\\.  If not, see <https://www\\.gnu\\.org/licenses/>\\.\n "
```
Comment (therealgrumpythomas) [inline .eslintrc.json:176]: Windows uses `\r\n` as EOL indicator. Wonder if developers on windows will have problems passing the linter. Darius will put his editor configuration into a [PR](https://github.com/themoeway/yomitan/pull/101/files#r1145575033). Maybe we can force the `\n` EOL in there as well?

**Review by therealgrumpythomas — COMMENTED** [review]
Looks good to me. I have some thought about windows compatibility. Maybe we can add an editorconfig file or vssettings file so everyone uses \n per default. Can be coordinated on a different issue

File: .eslintrc.json (hunk 1 above)
Comment (Aquafina-water-bottle) [reply .eslintrc.json:176]: Oh good catch. The array format seems like it tests for any of `\r|\n|\r\n` between the strings, so this is technically a regression. I'm for forcing that in the editorconfig if possible :pray:

Comment (djahandarie) [reply .eslintrc.json:176]: Let's do that in editorconfig :+1:

**Review by djahandarie — APPROVED** [review]
LGTM, agreed on editorconfig.

**Pull request merged**